pkg/kubelet/kubelet.go

```diff
@@ -1458,12 +1467,18 @@ func (kl *Kubelet) GetKubeletContainerLogs(podFullName, containerName, tail stri
 
 // GetBoundPods returns all pods bound to the kubelet and their spec
 func (kl *Kubelet) GetBoundPods() ([]api.BoundPod, error) {
-	return kl.pods, nil
+	kl.podLock.RLock()
+	defer kl.podLock.RUnlock()
+	var result []api.BoundPod
+	copy(kl.pods, result)
+	return result, nil
 }
 
 // GetPodFullName provides the first pod that matches namespace and name, or false
 // if no such pod can be found.
 func (kl *Kubelet) GetPodByName(namespace, name string) (*api.BoundPod, bool) {
+	kl.podLock.RLock()
+	defer kl.podLock.RUnlock()
```

Comment:
If it is truly necessary to copy in the GetBoundPods function, then it must also be unsafe to return a pointer in this function, as is done below.